src/plugins/intel_gpu/src/graph/loop.cpp

```diff
@@ -1003,16 +1003,16 @@ void loop_inst::set_memory_in_body_network(cldnn::network::ptr body_network,
                 const std::shared_ptr<cldnn::primitive_inst>& inst, memory::ptr mem) {
     if (inst->is_input()) {
         // in case where memory buffer has been over-allocated by shape predictor, memory layout might be unexpected shape.
-        // so memory layout needs to be re-interprete according to original layout.
+        // so memory layout needs to be re-interpret according to original layout.
         memory::ptr updated_mem = mem;
         layout impl_layout = inst->get_impl_params()->get_output_layout();
         OPENVINO_ASSERT(impl_layout.bytes_count() <= updated_mem->get_layout().bytes_count(),
                         "impl_params layout size(", impl_layout.to_short_string(),
                         ") should not exceed memory size(", updated_mem->get_layout().to_short_string(), ")");
-        if (impl_layout.bytes_count() < updated_mem->get_layout().bytes_count()) {
-            updated_mem = body_network->get_engine().reinterpret_buffer(*updated_mem, impl_layout);
-        }
-        body_network->set_input_data(inst->id(), updated_mem);
+        // Set need_to_check_memory_to_set to false to set output memory even if the input node has static shape,
+        body_network->set_input_data(inst->id(), updated_mem, false);
+        // Update impl_params.output_layouts[0] to updated_mem's layout
+        inst->update_shape();
```

Comment:
I am now curious, 
Couldn't the original issue be resolved simply by applying following change? 
![image](https://github.com/user-attachments/assets/3b488914-99ee-44fc-b902-d582619ae539)
<= at line 1007, set impl_layout  from the inst->get_input_layout() instead of output, if the input shape is static.

Reply:
Synced with @ahnyoung-paul  and will accept this for now